File: app/src/test/java/com/okuzawats/truth/UnitSubject.java
package com.okuzawats.truth;

import static com.google.common.truth.Truth.assertAbout;

import androidx.annotation.Nullable;

import com.google.common.truth.FailureMetadata;
import com.google.common.truth.Subject;

import kotlin.Unit;

public class UnitSubject extends Subject {
  public static UnitSubject assertThat(@Nullable Unit actual) {
    return assertAbout(employees()).that(actual);
  }

  public static Subject.Factory<UnitSubject, Unit> employees() {
    return UnitSubject::new;
  }

  private final Unit actual;
  private UnitSubject(FailureMetadata metadata, @Nullable Unit subject) {
    super(metadata, subject);
    this.actual = subject;
  }

  public void isUnit() {
    unit().isEqualTo(Unit.INSTANCE);
  }

  public void isNotUnit() {
    unit().isNotEqualTo(Unit.INSTANCE);
  }

  public Subject unit() {
    return check("name()").that(actual);
  }
}

File: app/src/test/java/com/okuzawats/truth/UnitSubject.kt
package com.okuzawats.truth

import com.google.common.truth.FailureMetadata
import com.google.common.truth.Subject
import com.google.common.truth.Subject.Factory
import com.google.common.truth.Truth

class UnitSubject private constructor(metadata: FailureMetadata, private val actual: Unit?) :
    Subject(metadata, actual) {

    fun isUnit() =  unit().isEqualTo(Unit)

    fun isNotUnit() = unit().isNotEqualTo(Unit)

    fun unit(): Subject {
        return check("name()").that(actual)
    }

    companion object {
        fun assertThat(actual: Unit?): UnitSubject {
            return Truth.assertAbout(employees()).that(actual)
        }

        fun employees(): Factory<UnitSubject, Unit?> {
            return Factory { metadata: FailureMetadata, subject: Unit? ->
                UnitSubject(
                    metadata,
                    subject
                )
            }
        }
    }
}
